modules/pafRtdProvider.js

```diff
@@ -0,0 +1,113 @@
+
+import { submodule } from '../src/hook.js';
+import { mergeDeep, isPlainObject, logMessage, deepSetValue, generateUUID } from '../src/utils.js';
+import { getGlobal } from '../src/prebidGlobal.js';
+import {config} from '../src/config.js';
+
+const SUBMODULE_NAME = 'paf';
+
+/**
+ *
+ * @param {Object} reqBidsConfigObj
+ * @param {function} callback
+ * @param {Object} rtdConfig
+ * @param {Object} userConsent
+ */
+function getBidRequestData(reqBidsConfigObj, onDone, rtdConfig, userConsent) {
+  logMessage('DEBUG(paf):', rtdConfig);
+  let idsAndPreferences;
+  const adUnits = (reqBidsConfigObj.adUnits || getGlobal().adUnits);
+
+  if (rtdConfig.params && rtdConfig.params.proxyHostName && window.PAF) {
+    idsAndPreferences = window.PAF.getIdsAndPreferences();
+    if (!idsAndPreferences) {
+      onDone();
+      return;
+    }
+
+    let transaction_ids = [];
+    for (var i=0; i < adUnits.length; i++) {
+      const uuid = generateUUID();
+      transaction_ids.push(uuid)
+      deepSetValue(adUnits[i], `ortb2Imp.ext.data.paf.transaction_id`, uuid)
+    }
+
+    logMessage('DEBUG(idsAndPreferences):', idsAndPreferences);
+    window.PAF.createSeed({proxyHostName: rtdConfig.params.proxyHostName, callback: function (seed) {setData(seed, rtdConfig, onDone);}}, transaction_ids)
```

Comment:
@OlivierChirouze @RomainLofaso 
I'm not an expert but I see that createSeed is using promise here https://github.com/prebid/paf-mvp-implementation/blob/ae8782f16f2aae49f79feb061ab67d0c0958e02a/paf-mvp-frontend/src/lib/paf-lib.ts#L454
Is it ok to call it this way ? (with a callback packed together in an object with the first argument, proxyHostName)

Reply:
Thanks for the heads up.
It is related to another PR on PAF: https://github.com/prebid/paf-mvp-implementation/pull/88/files/399fa630167948020d5b0b226da305713cfb3d47#diff-77c3a78d4eeb5e57d6a99735308b4e8982ca2de5077ac0755e8a619e4b2ed81cR137
The suggested change includes introducing this new `callback` member on the configuration object.
Now, it is true that `createSeed` itself is `async` but it just means that this `getBidRequestData` function will not wait for `createSeed` to complete before to return, which I guess is ok.